FWCore/Utilities/interface/ExceptionCollector.h

```diff
@@ -35,7 +35,7 @@ namespace edm {
     ExceptionCollector(std::string const& initialMessage);
     ~ExceptionCollector();
     bool hasThrown() const;
-    void rethrow() const;
+    [[noreturn]] void rethrow() const;
```

Comment:
The clang compilation warning collector missed
```
src/FWCore/Utilities/src/ExceptionCollector.cc:36:3: warning: function declared 'noreturn' should not return [-Winvalid-noreturn]
   36 |   }
      |   ^
1 warning generated.
```
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-730ac3/42230/buildClang.log

The warning is real, as the `rethrow()` function has a code path that returns (`nExceptions_ == 0`)
https://github.com/cms-sw/cmssw/blob/1637e96339168f5237472a371cccf275ecf45ee3/FWCore/Utilities/src/ExceptionCollector.cc#L30-L36

Reply:
@makortel thanks!